nixos/modules/services/cluster/kubernetes/kubelet.nix

```diff
@@ -6,6 +6,12 @@ let
   top = config.services.kubernetes;
   cfg = top.kubelet;
 
+  kubeletRbacUser =
+    if cfg.hostname == "" then
+      throw "kubelet.hostname must be set. This is probably caused by networking.hostName being empty."
```

Comment:
This should be done in an assertion (the module system assertion) and not halt the entire module system.